test/unit/component.js

```diff
@@ -82,142 +85,344 @@ describe('Component class', () => {
-          assert.deepEqual(model, {test: 'lol'});
+      describe('DOMEvents', () => {
+        beforeEach(() => {
+          component = Object.defineProperty(component, 'options', {
+            writable: true,
+          });
+        });
+
+        it('returns options.DOMEvents if it exists', () => {
+          component.options = {DOMEvents: 'test'};
+          assert.equal(component.DOMEvents, 'test');
+        });
+
+        it('returns an empty object if options.DOMEvents does not exist', () => {
+          component.options = {DOMEvents: null};
+          assert.deepEqual(component.DOMEvents, {});
         });
       });
 
-      it('calls fetchData if not passed data', () => {
-        const component = new Component({id: 1234});
-        component.fetchData = sinon.stub().returns(Promise.resolve({test: 'lol'}));
-        return component.setupModel().then((model) => {
-          assert.deepEqual(model, {test: 'lol'});
+      describe('events', () => {
+        beforeEach(() => {
+          component = Object.defineProperty(component, 'options', {
+            writable: true,
+          });
+        });
+
+        it('returns options.events if it exists', () => {
+          component.options = {events: 'test'};
+          assert.equal(component.events, 'test');
+        });
+
+        it('returns an empty object if options.events does not exist', () => {
+          component.options = {events: null};
+          assert.deepEqual(component.events, {});
         });
       });
-    });
 
-    describe('updateConfig()', () => {
-      it('delegates to updater', () => {
-        const component = new Component({id: 1234});
-        component.updater.updateConfig = sinon.spy();
-        component.updateConfig({test: 'lol'});
-        assert.calledWith(component.updater.updateConfig, {test: 'lol'});
+      describe('styles', () => {
+        it('returns styles for each component in manifest', () => {
+          component = new Component({
+            id: 1234,
+            options: {
+              product: {
+                styles: {
+                  button: {
+                    color: 'red',
+                  },
+                },
+              },
+            },
+          });
+          component.typeKey = 'product';
+          assert.deepEqual(component.styles, {product: {button: {color: 'red'}}});
+        });
+      });
+
+      describe('classes', () => {
+        it('returns classes for each component in manifest', () => {
+          component.typeKey = 'product';
+          assert.equal(component.classes.product.product, 'shopify-buy__product');
+          assert.equal(component.classes.option.option, 'shopify-buy__option-select');
+        });
+      });
+
+      describe('selectors', () => {
+        it('returns classes formatted as css selectors for each component in manifest', () => {
+          component.typeKey = 'product';
+          assert.equal(component.selectors.product.product, '.shopify-buy__product');
```

Comment:
Same as above regarding to where these values are coming from and how does the manifest affects these selections.